### ISSUE TITLE
LuxCore hangs on ending render in specific scene

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#308]
LuxCore version: v2.3rc1
OS: Windows 10

Scene: [Atmosphere_LuxCore_hang_on_ending_render.zip](https://github.com/LuxCoreRender/LuxCore/files/4298981/Atmosphere_LuxCore_hang_on_ending_render.zip)

Steps to reproduce:
* Start render
* Try to end it (either by letting it run over the halt condition, or by pressing escape)

It also hangs in viewport render when trying to end it.
I only tested Path CPU/RT Path CPU so far.

edit: It does not hang when I render with Path OpenCL on the GPU only.

**Comment by Dade916** [comment]
I should have fixed the problem. There were 2 problems:

1) The threads where trapped in an endless loop while tracing a ray on so large values where t + epsilon was still equal to t and the position on the ray was never advancing.

2) Embree was returning a ray intersection t out of the asked range [min_t, max_t] . This is technically a bug in Embree. I added a safety check.

Trying to work with so large numbers with 32bits floating point is probably prone to hundreds of problems like this one.

**Comment by Theverat** [comment]
It might be a good idea to find/calculate the maximum supportable scene size, and throw a warning or error before starting the render of a scene that exceeds it. Then we only have to find fixes or workarounds for scenes inside the supported range and can avoid costly extra checks for scenes outside of it. And users could be sure that valid scenes render or get a helpful error message in case their scene is too big.

However, I don't know if this is possible and how much work it would be, and it's probably low priority depending on these factors.
